torch/csrc/api/include/torch/detail/TensorDataContainer.h

```diff
@@ -225,6 +228,11 @@ AT_FORALL_SCALAR_TYPES_AND2(Half, BFloat16, TENSOR)
       fill_tensor(tensor);
       return tensor.to(options.device());
     } else if (is_tensor()) {
+      bool from_complex = tensor_.scalar_type() == at::kComplexFloat || tensor_.scalar_type() == at::kComplexDouble;
+      bool to_complex = options.dtype() == at::kComplexFloat || options.dtype() == at::kComplexDouble;
```

Comment:
same as above

Reply:
Does tensor options have `is_complex`?

Reply:
no but you can do `bool to_complex = at::empty({0}, options).is_complex()`

Reply:
fixed